auth/credentials/idtoken/idtoken.go

```diff
@@ -113,13 +113,20 @@ func NewCredentials(opts *Options) (*auth.Credentials, error) {
 	if err := opts.validate(); err != nil {
 		return nil, err
 	}
-	if b := opts.jsonBytes(); b != nil {
-		return credsFromBytes(b, opts)
-	}
-	if metadata.OnGCE() {
+	b := opts.jsonBytes()
+	if b == nil && metadata.OnGCE() {
 		return computeCredentials(opts)
+	} else if b == nil {
+		creds, err := credentials.DetectDefault(&credentials.DetectOptions{
+			Audience: opts.Audience,
+		})
+		if err != nil {
+			return nil, err
+		}
```

Comment:
Note that in the case of ADC and impersonated or external account credential types, the `creds` created here are discarded, then re-created in a new call to `DetectDefault` (with additional `DetectOptions` args) in `credsFromBytes`.

Reply:
I think those opts could just be all passed above.

Reply:
See questions above.

Reply:
I refactored to use `DetectDefault` in all non-compute cases, which simplifies the code a bit.